### PR TITLE
Login with DB

### PR DESCRIPTION
**Pull request by ytsheng**
Old User Login:
ezgif-2-412c9889bc

### CODE REVIEW

File: my-app/src/DataStore.jsx [hunk 1]
```diff
@@ -437,6 +448,40 @@ class DataStore {
...
+  getPersonByEmail(email, callback){
+    /**
+     * Gets the person with an email
+     * @param email {string} email to search by
+     * @return {Promise} promise for a {Person}
+     *          with the given email
+     */
+    if (this._persons.keys.length == 0) {
+      this.firestore.collection("persons")
+      .onSnapshot({/*config object*/}, (querySnapshot)=> {
```
Comment (bzreinhardt) [inline my-app/src/DataStore.jsx]: Do you want to use `onSnapShot()` or just `get()` here? Not sure why you need to listen to the database continually.

Comment (ytsheng) [reply my-app/src/DataStore.jsx]: yeah, that's better. Thanks!

**Comment by ytsheng** [comment]
Ben, please take a look at this again.

File: my-app/src/DataStore.jsx [hunk 2]
```diff
@@ -437,6 +448,40 @@ class DataStore {
     return Promise.resolve(Array.from(this._persons).filter(obj => obj[1].name === name).map(obj=>obj[1]));
   }
 
+  getPersonByEmail(email, callback){
+    /**
+     * Gets the person with an email
+     * @param email {string} email to search by
+     * @return {Promise} promise for a {Person}
```
Comment (bzreinhardt) [inline my-app/src/DataStore.jsx]: Does this necessarily return a promise?

Comment (bzreinhardt) [reply my-app/src/DataStore.jsx]: `getPersonsByEmail` works differently than `getPersonsByName` or `getPersonsById` - realizing that they should all work the same way. What I'd propose is:
- All return a promise
- All check local memory for the person and then fall back to checking the DB

**Comment by bzreinhardt** [comment]
Note that this still doesn't associate a person who's name is in the DB but not their email (ie me :-P)

**Comment by bzreinhardt** [comment]
@ytsheng fixed a couple of things - now works for me. Would you look at the changes and test it as well.

I've added to the issues the fact that logging out needs to clear data as well.

**Comment by ytsheng** [comment]
🍿

**Pull request merged**